arangod/Scheduler/Scheduler.cpp

```diff
@@ -208,28 +212,76 @@ Scheduler::~Scheduler() {
 }
 
 // do not pass callback by reference, might get deleted before execution
-void Scheduler::post(std::function<void()> const& callback) {
```

Comment:
Most of the calls to post pass `false` as last parameter (`isV8`). How about stripping this parameter and instead split out an additional function `postV8` that contains the V8 specific logic (no more ifs) and simply forwards a wrapped callback to `post`? This would not only make the calls to `post` not only simpler, but also clearer.